fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java

```diff
@@ -532,13 +534,23 @@ public List<Expression> visitNamedExpressionSeq(NamedExpressionSeqContext namedC
      * @return List of OrderKey
      */
     @Override
-    public List<OrderKey> visitQueryOrganization(QueryOrganizationContext ctx) {
+    public QueryOrganization visitQueryOrganization(QueryOrganizationContext ctx) {
         return ParserUtils.withOrigin(ctx, () -> {
+            List<OrderKey> orderKeys;
             if (ctx.sortClause().ORDER() != null) {
-                return visit(ctx.sortClause().sortItem(), OrderKey.class);
+                orderKeys = visit(ctx.sortClause().sortItem(), OrderKey.class);
             } else {
-                return ImmutableList.of();
+                orderKeys = ImmutableList.of();
             }
+            long limit = Long.MAX_VALUE;
```

Comment:
the limitClause definition above can ensure this limit must has a value.
If you keep the definition as before, I think this limit should be `Optional.empty()` and return originPlan without wrap a QueryOrganization.

Reply:
Thanks for your comments.
I have considered this solution. But I have some concerns.
1. If we use Optional, we should use Long object instead of primitive long for `limit`. This is not efficient.
2. with default value, the process of limit and offset could be unified.
For the above reasons, It is better to keep default value.

Reply:
You may have misunderstood, I means that local limit variable can set to Optional<Limit>, and if it's empty, we can return originPlan without build a LogicalLimit plan, otherwise we can build LogicalLimit and wrap the originPlan as child. In the LogicalLimit, the limit field also is `long limit`, no Optional.